### PR TITLE
If no md5 then check bytesize

### PR DESCRIPTION
**Pull request by orangeSi**

### CODE REVIEW

**Comment by orangeSi** [comment]
Sometimes files in xml have not md5 hash values, so consider to check the size of file with sizeInBytes.

**Review by glarue — CHANGES_REQUESTED** [review]
Look over my comments to see the suggestions makes sense. I've also added a `dev` branch that future pull requests should be targeted to (I've changed the target for these changes to that branch).

File: jgi-query.py [hunk 1]
```diff
@@ -394,11 +394,15 @@ def print_data(data, org_name, display=True):
...
-                    pass
+                url = i["url"]
+                dict_to_get[catID][index] = url
+                if url not in url_to_validate: url_to_validate[url] = {}
+                if "md5" in i:
+                    url_to_validate[url]['md5'] = i["md5"]
+                elif "sizeInBytes" in i:
+                    url_to_validate[url]['sizeInBytes'] = int(i["sizeInBytes"])
+                else:
+                    print(f"warn: no md5 or sizeInBytes, so the downloaded file maybe not intact for {org_name}.")
```
[the remark this answers is not in the document]
Comment (orangeSi) [reply jgi-query.py:406]: I think the situation(no md5 and no sizeInBytes in xml file) is alomost not happen, but if happened, maybe need print some message to inform user ?

```check_sizeInBytes()``` only works if have sizeInBytes in xml fils.

File: jgi-query.py [hunk 2]
```diff
@@ -581,7 +585,7 @@ def is_broken(filename, min_size_bytes=20, md5_hash=None):
         not os.path.isfile(filename) or
         os.path.getsize(filename) < min_size_bytes or 
         (is_xml(filename) and not filename.lower().endswith('xml') or
-        not check_md5(filename, md5_hash))
+        not check_md5(filename, md5_hash) or not check_sizeInBytes(filename, sizeInBytes))
```
[the remark this answers is not in the document]
Comment (orangeSi) [reply jgi-query.py:588]: Yes, better to rank-order them. At first I also think ```MD5``` maybe should have the  highest priority, and then ```sizeInBytes```  . 
In fact, the cost time of check MD5 is longer than the cost time of check ```sizeInBytes```,  but even the check MD5 of 1GB file only cost 2 sencods, So maybe can ignore their time cost difference.

**Comment by glarue** [comment]
@orangeSi thanks for your changes—I've integrated/added to this in the `dev` branch. I'll wait a bit to merge `dev` into `master`, so you can pull `dev` locally and check it out, and submit any further changes via PR.

**Pull request merged**